File: Lenguajes/python/suma.py
import time
import os

# Crea la carpeta 'resultados' si no existe
os.makedirs("resultados", exist_ok=True)

start = time.time()
resultado = sum(range(1, 10_000_000))
end = time.time()

tiempo = int((end - start) * 1000)

# 🔹 Imprime solo el tiempo (para `run.sh`)
print(tiempo)

# Guarda el resultado de la suma
with open("resultados/salida.txt", "w") as file:
    file.write(str(resultado))

# Guarda el tiempo de ejecución
with open("resultados/resultados.csv", "a") as file:
    file.write(f"Python,{tiempo}\n")

File: Lenguajes/java/suma.java
import java.io.*;

public class suma {
    public static void main(String[] args) {
        long start = System.currentTimeMillis();

        // Realiza la sumatoria
        long resultado = 0;
        for (int i = 1; i <= 10_000_000; i++) {
            resultado += i;
        }

        long end = System.currentTimeMillis();
        long tiempo = end - start;

        // Crea la carpeta 'resultados' si no existe
        File carpeta = new File("resultados");
        if (!carpeta.exists()) {
            carpeta.mkdirs();
        }

        // Guarda la suma en resultados/salida.txt
        try (FileWriter resultadoFile = new FileWriter("resultados/salida.txt")) {
            resultadoFile.write(String.valueOf(resultado));
        } catch (IOException e) {
            e.printStackTrace();
        }

        // Guarda el tiempo en resultados/resultados.csv
        try (FileWriter tiempoFile = new FileWriter("resultados/resultados.csv", true)) {
            tiempoFile.write("Java," + tiempo + "\n");
        } catch (IOException e) {
            e.printStackTrace();
        }

        System.out.println(tiempo);
    }
}
